### ISSUE TITLE
eol of codeblock is not aligned

### ISSUE THREAD

**Issue by sshelll** [OXY2DEV/markview.nvim#63]
**Describe the bug**
The codeblock displays in a wrong way.
My config is:
```lua
    code_blocks = {
        enable = true,
        style = "language",
        position = "overlay",

        hl = "MarkviewCodeblock",

        min_width = 100,
        pad_char = " ",
        pad_amount = 2,

        language_direction = "right",
        sign = false,
    }
```

**Screenshots**
![image](https://github.com/user-attachments/assets/559c1233-78db-491a-937a-91a1027243c0)

The json is:
```json
{
    "_id": "10001_9",
    "best": {
        "0": {
            "_id": "1",
            "s3Key": "1",
            "zoneId": 10001,
            "partition": 0,
            "offset": 8878339,
            "totalPartition": 9,
            "bytes": 1010248,
            "timestamp": "1721995277265924191",
            "s3Region": "1",
            "s3Bucket": "1",
            "filePath": "1",
            "writeMode": "s3"
        }
    }
}
```

**Neovim version**
0.10.0

**Additional context**
MacOS 14.5
Alacritty 0.13.2
Tmux 3.4
JetBrainsMono Nerd Font

**Comment by OXY2DEV** [comment]
This is caused by the `<Tab>` character.

Sometimes if you copy & past text from a different file the plugin doesn't know how many characters a line has(which is why every line is messed up by a different amount).

Unfortunately, this makes it very hard to fix since there doesn't seem to be any function that works consistently for lines containing `<Tab>` in them.

So, unfortunately, I currently can't fix this issue.

**Comment by sshelll** [comment]
Thanks for your explain, but the json dosen't contain any `<Tab>` 🧐.

Actually I think this bug is caused by the quotation mark: `"`.

Here's a simple demo:
```json
{
    "num_field": 100,
    "num_list": [1, 2],
    "str_field": "str",
    "str_list": ["1", "2"]
}
```

![image](https://github.com/user-attachments/assets/762ad104-52e5-4b22-ac39-c880852d97b2)

Hope this demo helps~ @OXY2DEV

**Comment by OXY2DEV** [comment]
> Thanks for your explain, but the json dosen't contain any `<Tab>` 🧐.
> 
> Actually I think this bug is caused by the quotation mark: `"`.
> 
> Here's a simple demo:
> 
> ```json
> {
>     "num_field": 100,
>     "num_list": [1, 2],
>     "str_field": "str",
>     "str_list": ["1", "2"]
> }
> ```
> 
> ![image](https://private-user-images.githubusercontent.com/49158400/352808568-762ad104-52e5-4b22-ac39-c880852d97b2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIxNjg1MTcsIm5iZiI6MTcyMjE2ODIxNywicGF0aCI6Ii80OTE1ODQwMC8zNTI4MDg1NjgtNzYyYWQxMDQtNTJlNS00YjIyLWFjMzktYzg4MDg1MmQ5N2IyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI4VDEyMDMzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIxODVjZjMwN2VmOGFkMDIxOGU1OWUzZDBhYzdkMmU0MjQyNzliYTJiNmM0Y2MyMjAyM2NmNzBjZGVlMjAxYTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.HzyahaR3uSHgJIN9CUggCdizS1Cee8M_DtKbNxMCiJs)
> 
> Hope this demo helps~ @OXY2DEV

That makes it a lot harder to fix.
Oh well, more work I guess.

**Comment by sshelll** [comment]
I'm not familiar with the nvim treesitter, but I'm curious about why are those quotation marks hidden?

I think the easiest way to solve this problem is to leave those marks in their original positions -- If markview.nvim hides these marks deliberately.

**Comment by OXY2DEV** [comment]
Markview doesn't do that(cause there's literally no way to detect " marks).

You most likely have a `Tree-sitter` parser for json(or something similar).

You probably don't use `conceallevel=2` so you never noticed it. But that's most likely what's happening.

**Comment by sshelll** [comment]
I do have a treesitter parser for json:
`:TSInstallInfo` -> `json [✓] installed`

And I've tried to add this config when setting up, it's still not working:
```lua
options = {
    on_enable = {
        conceallevel = 2,
        concealcursor = "n"
    },
    on_disable = {
        conceallevel = 0,
        concealcursor = ""
    },
}
```
I wonder if the code blocks with the given json displays in a right way on your device? Or it only happens on mine?

PS: I Love the UI of markview, if this issue in not planned to fix recently, I'll close it for now. Becasue I've set the bg of code block to 'none', it looks good enough to me.
